### PR TITLE
Feat: Read the content from SFTP file and load the data into Analytics Database Tables

### PR DESCRIPTION
**Pull request by puru-khedre**
Issue: #4 

A new feature is added to the `oms-bi` component, to fetch the CSV files from the SFTP location and then Load the data into the **Analytics Database**

### Implementation
* Leverage the `SystemMessage` to read the files from the `SFTP` location and then move them to the archive folder at sftp
* Using the `EntityDataLoader` to load the contents of the CSV file into the Analytics DB
---

This Pull request includes the following changes
* Define a `consume#AnalyticsDataSync` service to consume the file that is read from the SFTP
* Write data to define the `SystemMessageType` & `ServiceJob` records

### CODE REVIEW

File: service/co/hotwax/bi/SystemMessageServices.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<services xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="http://moqui.org/xsd/service-definition-3.xsd">
+
+    <service verb="consume" noun="AnalyticsDataSync">
+        <implements service="org.moqui.impl.SystemMessageServices.consume#SystemMessage"/>
+        <actions>
+
+            <entity-find-one entity-name="moqui.service.message.SystemMessage" value-field="systemMessage"/>
+            <set field="fileText" from="ec.resource.getLocationReference(systemMessage.messageText).getText()"/>
+            <set field="edl" from="ec.entity.makeDataLoader()"/>
```
Comment (diveshdut) [inline service/co/hotwax/bi/SystemMessageServices.xml]: edl is not used in the code

File: data/SftpReadDemoData.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<entity-facade-xml type="demo">
+    <moqui.service.message.SystemMessageType
+            systemMessageTypeId="AnalyticsDataSync"
```
Comment (diveshdut) [inline data/SftpReadDemoData.xml]: Let's rename it to "readOLAPData"

File: data/SftpReadDemoData.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            description="Upload the CSV file into Analytics Database"
```
Comment (diveshdut) [inline data/SftpReadDemoData.xml]: Reads the CSV file from SFTP and stores the data in entity which is defined in CSV file.

File: data/SftpReadDemoData.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            receiveFilePattern=".*\.csv"
+            sendPath="${contentRoot}/DataWarehouse"
+            receiveMovePath="/home/${sftpUsername}/hotwax/DataWarehouse/archive"
+            receivePath="/home/${sftpUsername}/hotwax/DataWarehouse"
+            receiveResponseEnumId="MsgRrMove"
+    />
+
+    <moqui.service.job.ServiceJob
+            jobName="poll_SystemMessageFileSftp_AnalyticsDataSync"
+            description="poll for the Analytics Data files to SFTP location"
```
Comment (diveshdut) [inline data/SftpReadDemoData.xml]: Polls to see if there is any CSV file so that data can be stored in OLAP database.

**Review by diveshdut — CHANGES_REQUESTED** [review]

**Review by diveshdut — APPROVED** [review]

File: data/SftpReadDemoData.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<entity-facade-xml type="demo">
+    <moqui.service.message.SystemMessageType
+            systemMessageTypeId="AnalyticsDataSync"
+            description="Upload the CSV file into Analytics Database"
+            consumeServiceName="co.hotwax.bi.SystemMessageServices.consume#AnalyticsDataSync"
```
Comment (puru-khedre) [inline data/SftpReadDemoData.xml]: @diveshdut sir, Should we also consider changing the name of this consume service and service job?

File: service/co/hotwax/bi/SystemMessageServices.xml (hunk 1 above)
Comment (puru-khedre) [reply service/co/hotwax/bi/SystemMessageServices.xml]: Removed unused code

File: data/SftpReadDemoData.xml (hunk 2 above)
Comment (puru-khedre) [reply data/SftpReadDemoData.xml]: Rename to `ReadOLAPData`. I have noticed a pattern where IDs are written in PascalCase

File: data/SftpReadDemoData.xml (hunk 4 above)
Comment (puru-khedre) [reply data/SftpReadDemoData.xml]: updated

**Comment by puru-khedre** [comment]
This PR resolves #4 issue

**Pull request merged**